File: meta/addPvalue.R
#!/usr/bin/env Rscript

sink(stderr())
library(data.table)

args <- commandArgs(T)
s <- args[1]
infile <- args[2]

descriptives <- fread("descriptives_cols.tsv")
beta_col <- descriptives[ study == s, beta ]
se_col <- descriptives[ study == s, se ]

results <- fread(cmd=paste("zcat", infile))
results[, P:=2*pnorm(-abs(get(beta_col)/get(se_col)))]

sink()
write.table(format(results, digits=5), "", row.names=F, quote=F, sep="\t")

File: meta/addPvalue.jl
#!/usr/bin/env julia

using Distributions

const d = Normal()
const study = ARGS[1]

beta_col = String
se_col = String

for line in eachline("descriptives_cols.tsv")
    cols = split(line)
    if (cols[1] == study)
        global beta_col = cols[10]
        global se_col = cols[11]
        break
    end
end


firstline = true
beta_idx = 0::Integer
se_idx = 0::Integer

for line in eachline()
    cols = split(line)
    if (firstline)
        global beta_idx = findfirst(isequal(beta_col), cols)
        global se_idx = findfirst(isequal(se_col), cols)
        println(line * "\tP")
        global firstline = false
    else
        P = try
            beta = parse(Float64, cols[beta_idx])
            se = parse(Float64, cols[se_idx])
            2*cdf(d, -abs(beta/se))
        catch e
            NaN
        end
        println(line, "\t",  round(P, sigdigits=5))
    end
end
